### PR TITLE
Base container

### PR DESCRIPTION
**Pull request by drivera-armedia**
Please review so we can proceed to testing

### CODE REVIEW

**Review by JonHolman — DISMISSED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1 +1,68 @@
...
+RUN go install -v "${SRC}/cmd/grafana-reporter@v${VER}"
+RUN cp -vi "${GOPATH}/bin/grafana-reporter" "/usr/local/bin"
+
+#
+# Copy the main executable over
+#
+COPY /startup.sh /
+RUN chmod ug+rwx,o-rwx /startup.sh
+
+USER        ${UID}
```
Comment (fabricetriboix) [inline Dockerfile:64]: Does it have to run as root? One of the best practices is to avoid running containerized apps as root...

File: Dockerfile [hunk 2]
```diff
@@ -1 +1,68 @@
...
+EXPOSE      8686
+VOLUME      [ "/templates" ]
+SHELL       [ "/bin/bash", "-c" ]
```
Comment (fabricetriboix) [inline Dockerfile]: This is the first time I am seeing this `SHELL` instruction... What does it do?

File: startup.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
```
Comment (fabricetriboix) [inline startup.sh:1]: I always put the first command in a bash script to be `set -eu -o pipefail`, this is defensive programming.

File: startup.sh [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+# -proto string
+#       Grafana Protocol. Change to 'https://' if Grafana is using https. Reporter will still serve http. (default "http://")
+GRAFANA_SSL="${GRAFANA_SSL,,}"
+case "${GRAFANA_SSL}" in
+	1 | y | yes | true | t | on ) GRAFANA_SSL="true" ;;
+	0 | n | no | false | f | off ) GRAFANA_SSL="false" ;;
+	* ) GRAFANA_SSL="false" ;;
+esac
+SSL_PROTO="http://"
```
Comment (fabricetriboix) [inline startup.sh:42]: The variable name is confusing, as the protocol could be either http or https. Maybe just `PROTO`? Also, is it really useful to do all that stuff given that SSL will be performed by Istio?

File: Dockerfile (hunk 1 above)
Comment (drivera-armedia) [reply Dockerfile:64]: Probably not, I'll look into this and modify accordingly - just duplicated what the original had.

File: Dockerfile (hunk 2 above)
Comment (drivera-armedia) [reply Dockerfile]: Declares the default shell to be used when someone opens a... ._shell_ ... to the container without specifying the executable.

Can probably left out ... though we should consider perhaps standardizing on /bin/bash for all containers? (the default is /bin/sh).

File: startup.sh (hunk 4 above)
Comment (drivera-armedia) [reply startup.sh:42]: Yes, this can probably be improved...

File: Dockerfile (hunk 2 above)
Comment (fabricetriboix) [reply Dockerfile]: Possibly... What's your take on this issue @JonHolman ?

Comment (drivera-armedia) [reply Dockerfile]: For the record: I don't feel strongly about this setting either way, which is why the other images don't include it... I just added it here b/c it's transplanted from the developer's example Dockerfile.

File: startup.sh (hunk 4 above)
Comment (drivera-armedia) [reply startup.sh:42]: @fabricetriboix I remember now why I chose this name: so it'd be consistent with other settings used for other Grafana images when declaring ENV values at instantiation time.

File: Dockerfile (hunk 2 above)
Comment (JonHolman) [reply Dockerfile]: I agree that we should be consistent.  I don't know if this needs to be set.  Can we go without it until we see a reason to add it (then add it to all, or at a base image level?).

Comment (drivera-armedia) [reply Dockerfile]: Sure. No sweat.

File: startup.sh (hunk 4 above)
Comment (fabricetriboix) [reply startup.sh:42]: Being consistent is good, and being readable is good. It looks like there is a balance to be found here, so I'll just trust you to make the right decision. It's a minor issue anyway.

Comment (drivera-armedia) [reply startup.sh:42]: Yes. I found a 2nd setting that also needed fixing - SSL_CHECK - so now they're both named "GRAFANA_SSL" and "GRAFANA_SSL_CHECK". This makes them easy to identify in environment settings.

File: startup.sh (hunk 3 above)
Comment (fabricetriboix) [reply startup.sh:1]: @drivera-armedia Any comment on this?

File: Dockerfile (hunk 1 above)
Comment (fabricetriboix) [reply Dockerfile:64]: Any progress on this?

File: startup.sh (hunk 3 above)
Comment (drivera-armedia) [reply startup.sh:1]: For this script it's not really relevant as all it does is sanitize some variables and set others, so there's no "commands" really that can fail, and the final command is an exec so if that fails, we're still crashing. But I have no objection to this. I'll test it and if it works fine, then I'll leave it in.

File: Dockerfile (hunk 1 above)
Comment (drivera-armedia) [reply Dockerfile:64]: Will test it later today. I don't foresee an issue other than perhaps having to change some permissions in designated work areas.

Comment (fabricetriboix) [reply Dockerfile:64]: @drivera-armedia Have you been able to test it, so we can close this ticket?

Comment (drivera-armedia) [reply Dockerfile:64]: Yes. It worked fine.

**Review by fabricetriboix — APPROVED** [review]

**Review by JonHolman — APPROVED** [review]

**Pull request merged**